perfkitbenchmarker/providers/gcp/gce_virtual_machine.py

```diff
@@ -283,11 +283,23 @@ def _GenerateCreateCommand(self, ssh_keys_path):
       cmd.flags['machine-type'] = self.machine_type
     cmd.flags['tags'] = 'perfkitbenchmarker'
     cmd.flags['no-restart-on-failure'] = True
-    cmd.flags['metadata-from-file'] = 'sshKeys=%s' % ssh_keys_path
-    metadata = ['owner=%s' % FLAGS.owner]
-    for key, value in self.boot_metadata.iteritems():
-      metadata.append('%s=%s' % (key, value))
-    cmd.flags['metadata'] = ','.join(metadata)
+
+    metadata_from_file = {'sshKeys': ssh_keys_path} if ssh_keys_path else {}
+    for pair in FLAGS.vm_metadata_from_file:
+      key, value = pair.split('=')
+      metadata_from_file.setdefault(key, value)
+    cmd.flags['metadata-from-file'] = ','.join([
+        '%s=%s' % (key, value) for key, value in metadata_from_file.iteritems()
```

Comment:
flake8 complains here about key and value being redefined (it was defined once in the for loop above, and it's being redefined here in the list comprehension). Please rename these.